src/game/SaveLoadGame.cc

```diff
@@ -1132,6 +1138,85 @@ static void SaveMercProfiles(HWFILE const f)
 	}
 }
 
+const char * IMPSavedProfileCreateFilename(const char *nickname_cstring)
+{
+	char *profile_filename = (char *)malloc(strlen(GCM->getSavedGamesFolder().c_str()) + 32);
+	sprintf(profile_filename, "%s/mercprofile.%s",GCM->getSavedGamesFolder().c_str(), nickname_cstring);
+	return profile_filename;
+}
+
+bool IMPSavedProfileDoesFileExist(const char *nickname)
+{
+	const char *profile_filename = IMPSavedProfileCreateFilename(nickname);
+	bool fexists = Fs_exists(profile_filename);
+	free((void*)profile_filename);
+	return fexists;
+}
+
+SGPFile* const IMPSavedProfileOpenFileForRead(const char *nickname)
+{
+	if (!IMPSavedProfileDoesFileExist(nickname)) return (SGPFile *)-1;
+	const char *profile_filename = IMPSavedProfileCreateFilename(nickname);
+	SGPFile *f = FileMan::openForReading(profile_filename);
+	free((void*)profile_filename);
+	return f;
+}
+
+SGPFile* const IMPSavedProfileOpenFileForWrite(const char *nickname)
+{
+	const char *profile_filename = IMPSavedProfileCreateFilename(nickname);
+	SGPFile *f = FileMan::openForWriting(profile_filename, true);
+	free((void*)profile_filename);
+	return f;
+}
+
+int IMPSavedProfileLoadMercProfile(const char *nickname)
+{
+	if (!IMPSavedProfileDoesFileExist(nickname)) return -1;
+	SGPFile *f = IMPSavedProfileOpenFileForRead(nickname);
+	MERCPROFILESTRUCT profile_saved;
+	FileRead(f, &profile_saved, sizeof(MERCPROFILESTRUCT));
+	FileClose(f);
+	int voiceid = profile_saved.ubSuspiciousDeath;
+	MERCPROFILESTRUCT& profile_new = gMercProfiles[PLAYER_GENERATED_CHARACTER_ID + voiceid];
+	profile_new = profile_saved;
+	profile_new.bMercStatus = MERC_OK;
+	return voiceid;
+}
+
+void IMPSavedProfileLoadInventory(const char *nickname, SOLDIERTYPE *pSoldier)
+{
+	if (!IMPSavedProfileDoesFileExist(nickname)) return;
+	if (!pSoldier) return;
+
+	SGPFile *f = IMPSavedProfileOpenFileForRead(nickname);
+	FileSeek(f, sizeof(MERCPROFILESTRUCT), FILE_SEEK_FROM_START);
+	FileRead(f, pSoldier->inv, sizeof(OBJECTTYPE) * NUM_INV_SLOTS);
+	FileClose(f);
+}
+
+void SaveIMPPlayerProfiles()
+{
+	#define LASTIMPPROFILE NPC169 // (PLAYER_GENERATED_CHARACTER_ID+5)
```

Comment:
`(PLAYER_GENERATED_CHARACTER_ID+5)` is correct, isn't it? Why don't we use it? We don't even need this #define.

Reply:
Or if we are planning for future (for UB - there is a new IMP profile ID), this define should be placed in Solder_Profile.h.

Reply:
@wvx do you remember why you tried both constants?

Reply:
This looks like I missed some of the split for NPC save/load. The first define would be just for IMP save/load and then it was extended to include NPCs so that the scenario would more closely match the source story.